src/kmeans.jl

```diff
@@ -116,35 +146,32 @@ function _kmeans!(
         else
             # if only a small subset is affected, only compute for that subset
             affected_inds = findall(to_update)
-            dmat_p = pairwise(distance, centers[:, affected_inds], X)
-            dmat[affected_inds, :] .= dmat_p
+            pairwise!(view(dmat, affected_inds, :), distance,
+                      view(centers, :, affected_inds), X)
         end
 
         # update assignments
-
-        update_assignments!(dmat, false, assignments, costs, counts, to_update, unused)
-
+        update_assignments!(dmat, false, assignments, costs, counts,
+                            to_update, unused)
         num_affected = sum(to_update) + length(unused)
 
         # compute change of objective and determine convergence
         prev_objv = objv
-        objv = w === nothing ? sum(costs) : dot(w, costs)
+        objv = weights === nothing ? sum(costs) : dot(weights, costs)
         objv_change = objv - prev_objv
 
         if objv_change > tol
             @warn("The objective value changes towards an opposite direction")
-        end
-
-        if abs(objv_change) < tol
+        elseif abs(objv_change) < tol
             converged = true
         end
 
-        # display iteration information (if asked)
-
+        # display information (if required)
         if displevel >= 2
-            @printf("%7d %18.6e %18.6e | %8d\n", t, objv, objv_change, num_affected)
+            @printf("%7d %18.6e %18.6e | %8d\n", t, objv, objv_change,
+                    num_affected)
         end
-    end
+    end # end while
```

Comment:
This is not the common Julia style convention AFAIK.

Reply:
I'll remove that along with the refactoring notes